### ISSUE TITLE
Perfomance optimization

### ISSUE THREAD

**Issue by OmarWagih1** [scrapinghub/dateparser#624]
Hello, I'm Omar Wagih, a student from Cairo University and i was interested in the "Performance optimization" task in GSoC.

As mentioned in issue #457 , there are some parts of the library that slows over time, a PR was issued but it still isn't integrated in master, @Gallaecio mentioned that we could create an issue where we can list different performance improvements that could be implemented as part of this GSoC idea. 

does anyone have any suggestions?

**Comment by Gallaecio** [comment]
I got these suggestions from @noviluni :

* As we implemented the PARSERS settings (PR: #603) now we can easily change the order of the parsers. It could be a good idea to locate those faster parsers first. To do it we should measure which parsers need more time and which need less time and order them according to this (on the other hand, if a parser is really uncommon maybe it shouldn't be in the first positions).
* Avoid parsing empty strings (or strings containing only spaces, etc.). I think there is already a PR addressing this, but it's not merged.
* Improve the tokenizer to use built-in functions etc. to know if its a char a digit or not instead of checking if it's inside a list (a string in fact) including all the letters (I don't know if this is possible or if this will affect the performance).
